pkg/route/engine.go

```diff
@@ -634,6 +634,20 @@ func (engine *Engine) recv(ctx *app.RequestContext) {
 	}
 }
 
+// HandleContext Re-enter a context that has been rewritten.
+// This can be done by setting ctx.Request.SetRequestURI() to new target and response will be reset.
+// Disclaimer: Maybe loop self to death with this, use wisely.
+func (engine *Engine) HandleContext(c context.Context, ctx *app.RequestContext) {
+	old := ctx.GetIndex()
+	ctx.Response.Reset()
+	ctx.Request.ResetURI()
```

Comment:
.ResetURI() will change req.isTLS to false(if it is true before) - this may cause problem in TLS mode?

Reply:
thanks

Reply:
No, because it will re-parsed.